evalml/pipelines/utils.py

```diff
@@ -192,13 +192,13 @@ def generate_pipeline_code(element):
     base_string = "\nclass {0}({1}):\n" \
                   "\tcomponent_graph = [\n\t\t{2}\n\t]\n" \
                   "{3}" \
-                  "\nparameters = {4}\n" \
+                  "\nparameters = json.loads(\"\"\"{4}\"\"\")\n" \
                   "pipeline = {0}(parameters)" \
                   .format(element.__class__.__name__,
                           element.__class__.__bases__[0].__name__,
                           component_graph_string,
                           pipeline_string,
-                          json.dumps(element.parameters, indent='\t').replace('null', 'None'))
+                          json.dumps(element.parameters, indent='\t'))
```

Comment:
not-blocking as it doesn't relate to this fix but it does relate to your question about which types we should handle. I think we should at least raise an actionable user-friendly exception if for whatever reason the parameters are not json serializable. Right now, the user gets a messy stack trace. 

For example, I used a numpy array to specify the `features_to_extract` parameter  instead of a list in the code gen demo. This is a corner-case for sure but the date time featurizer will work if you use a numpy array instead of a list.

```python
pipeline_instance = CustomPipeline(parameters={"Imputer": {"numeric_impute_strategy": "median"},
                                               "DateTime Featurization Component": {"features_to_extract": np.array(["year"])}})
code = generate_pipeline_code(pipeline_instance)
exec(code)
```

Stack trace:
```
~/sources/evalml/evalml/pipelines/utils.py in generate_pipeline_code(element)
    199                           component_graph_string,
    200                           pipeline_string,
--> 201                           json.dumps(element.parameters, indent='\t'))
    202     code_strings.append(base_string)
    203     return "\n".join(code_strings)

~/miniconda3/envs/evalml/lib/python3.8/json/__init__.py in dumps(obj, skipkeys, ensure_ascii, check_circular, allow_nan, cls, indent, separators, default, sort_keys, **kw)
    232     if cls is None:
    233         cls = JSONEncoder
--> 234     return cls(
    235         skipkeys=skipkeys, ensure_ascii=ensure_ascii,
    236         check_circular=check_circular, allow_nan=allow_nan, indent=indent,

~/miniconda3/envs/evalml/lib/python3.8/json/encoder.py in encode(self, o)
    199         chunks = self.iterencode(o, _one_shot=True)
    200         if not isinstance(chunks, (list, tuple)):
--> 201             chunks = list(chunks)
    202         return ''.join(chunks)
    203 

~/miniconda3/envs/evalml/lib/python3.8/json/encoder.py in _iterencode(o, _current_indent_level)
    429             yield from _iterencode_list(o, _current_indent_level)
    430         elif isinstance(o, dict):
--> 431             yield from _iterencode_dict(o, _current_indent_level)
    432         else:
    433             if markers is not None:

~/miniconda3/envs/evalml/lib/python3.8/json/encoder.py in _iterencode_dict(dct, _current_indent_level)
    403                 else:
    404                     chunks = _iterencode(value, _current_indent_level)
--> 405                 yield from chunks
    406         if newline_indent is not None:
    407             _current_indent_level -= 1

~/miniconda3/envs/evalml/lib/python3.8/json/encoder.py in _iterencode_dict(dct, _current_indent_level)
    403                 else:
    404                     chunks = _iterencode(value, _current_indent_level)
--> 405                 yield from chunks
    406         if newline_indent is not None:
    407             _current_indent_level -= 1

~/miniconda3/envs/evalml/lib/python3.8/json/encoder.py in _iterencode(o, _current_indent_level)
    436                     raise ValueError("Circular reference detected")
    437                 markers[markerid] = o
--> 438             o = _default(o)
    439             yield from _iterencode(o, _current_indent_level)
    440             if markers is not None:

~/miniconda3/envs/evalml/lib/python3.8/json/encoder.py in default(self, o)
    177 
    178         """
--> 179         raise TypeError(f'Object of type {o.__class__.__name__} '
    180                         f'is not JSON serializable')
    181 

TypeError: Object of type ndarray is not JSON serializable
```

Reply:
Great point! I'll add some code to make this more user-friendly